### ISSUE TITLE
How to edit the OSEM.m to transform the sinogram I have  to image directly ?

### ISSUE THREAD

**Issue by Joker-Yision** [villekf/OMEGA#54]
Because I am not familiar with matlab code， there are a lot of code I can't understand absolutely. Now , I have the ready-made sinogram，1501 × 471 from the PET simulation，in order to transform it to the reconstructed image by this OSEM algorithm. So，how can I do to reach it? edit the OSEM.m? How to do it? I have been trapped with it for a long time.
thank you!
best wishes!

**Comment by villekf** [comment]
Have you looked at the examples? Basically you'll need two steps: define the scanner, sinogram and reconstruction parameters and input the measurement data (sinogram) into `options.SinM`.

A simple example is here: [https://github.com/villekf/OMEGA/blob/master/main-files/PET_main_gateExampleSimple.m](https://github.com/villekf/OMEGA/blob/master/main-files/PET_main_gateExampleSimple.m). For Python there is a similar example here: [https://github.com/villekf/OMEGA/blob/master/source/Python/gate_PET_simpleExample.py](https://github.com/villekf/OMEGA/blob/master/source/Python/gate_PET_simpleExample.py)

**Comment by Joker-Yision** [comment]
Thank you for you answer！It is really be helpful to me. But now except for the parameters about the scanner and reconstruction, I  just have the sinogram（from geant4 not GATE) processed by ourselves(so now the sinogram is .mat type not a .root),  and I have looked at the exmple you informed. Whether my sinogram data can be identified and used or not? If so, I have to search for other ways to get my date of GATE root type.
Thanks again！

**Comment by villekf** [comment]
Yes, you can input your sinogram from a mat-file. Simply input it into `options.SinM` variable and then make sure that all the scanner, sinogram and reconstruction properties match your simulated scanner.

**Comment by Joker-Yision** [comment]
Thank you for your answer! Now, I have done the steps you informed above.(define the scanner in my program of matlab like the example, and reconstruction parameters, very simple because I just want to reconstructed a 2D image from a ring,  and input my sinogram.mat in options.SinM)

But unfortunately, errors "projector_mex can't be identified" like the figure following, I don't understand this parameter and how to solve it.

thanks again!
![屏幕截图 2024-12-19 203236](https://github.com/user-attachments/assets/804a76ae-78a6-4af0-a7f5-af38ccd34573)

**Comment by villekf** [comment]
Did you compile the mex-files ([https://omega-doc.readthedocs.io/en/latest/installation.html](https://omega-doc.readthedocs.io/en/latest/installation.html))?

**Comment by Joker-Yision** [comment]
Thank you！How about the errors following, it can't find the file('Cylindrical_bPET_cylpet_bPET_new_sinograms_combined_static_1501x471x1_span1.mat') and actually I do know what't that and the file must contain what information.

Thank you for you answer! I sincerely appreciate your taking the trouble to help me.
![屏幕截图 2024-12-19 2](https://github.com/user-attachments/assets/0b7631e7-2127-42bf-bd83-400953cd3489)

**Comment by villekf** [comment]
Did you input your sinogram into `options.SinM` before running the reconstruction?

**Comment by villekf** [comment]
Then it disappears somewhere along the way. 

You can try one or some of the following: 
a) Debug the case manually to see at which points the sinogram still exists and where it disappears
b) Rename your sinogram variable into `raw_SinM` and save it as the mat-file shown in the error message
c) Send me the main-file and sinogram data for testing

**Comment by Joker-Yision** [comment]
Thank you for your answer! I will try the first two， if both of them don‘t work out， I will send you the data. 
Thank you again!

**Comment by Joker-Yision** [comment]
Dear villekf :
&nbsp; I just do the operations like this , then run the bPET_configure(brain PET, and it’s the initial stages of a project , so you can see the very simple configure in the code).&nbsp;

Now I just simulate a single ring containing 942 1mm×1mm×30mm crystals about the PET scanner, the diameter is 30cm, no corrections. And I just want to get a reconstructed 2D image.


There are many errors in my code, maybe.It may be the lack of code due to my unfamiliarity with code and image reconstruction. But I did spend a lot of time on it without solving the problem.


&nbsp;So thank you very much for taking time out of your busy schedule to answer my questions!


Best wishes!&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
From Joker-Yision






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "villekf/OMEGA"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月19日(星期四) 晚上9:33
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [villekf/OMEGA] How to edit the OSEM.m to transform the sinogram I have  to image directly ? (Issue #54)





 
Then it disappears somewhere along the way.
 
You can try one or some of the following:
 a) Debug the case manually to see at which points the sinogram still exists and where it disappears
 b) Rename your sinogram variable into raw_SinM and save it as the mat-file shown in the error message
 c) Send me the main-file and sinogram data for testing
